core/src/main/java/jenkins/security/QueueItemAuthenticatorMonitor.java

```diff
@@ -110,15 +122,17 @@ public void onLeft(Queue.LeftItem li) {
                 return;
             }
 
-            // XXX This can probably be done better, unsure what the expected priority order is
+            if (!isQueueItemAuthenticatorPresent() || !isQueueItemAuthenticatorConfigured()) {
+                // ignore while this is not configured
+                return;
+            }
+
+            // XXX This can probably be done better, unsure what the expected priority order should be
             String displayName = li.getDisplayName();
             if (displayName == null && li.task != null) {
                 displayName = li.task.getFullDisplayName();
             }
-            Queue.Executable executable = li.getExecutable();
-            if (displayName == null && executable != null) {
-                displayName = executable.toString();
-            }
+            // Do not need to check Executable as alternative while Queue#onStartExecuting(Executor) is called before Task#createExecutable() from Executable.
```

Comment:
I wondered why executable wasn't set when I wrote the first PR, this is the reason. Also the reason we have two different listeners. One can tell the authentication, the other can tell the build number.